utils/sample_qc.py

```diff
@@ -0,0 +1,487 @@
+    :param bool apply_hard_filters: Whether to apply standard GAKT default site hard filters: QD >= 2, FS <= 60 and MQ >= 30
+    :param float ld_r2: Minimum r2 to keep when LD-pruning (set to `None` for no LD pruning)
+    :param bool filter_exome_low_coverage_regions: If set, only high coverage exome regions (computed from gnomAD are kept)
+    :param list of str high_conf_regions: If given, the data will be filtered to only include variants in those regions
+    :return: Filtered MT
+    :rtype: MatrixTable
+    """
+    logger.info("Creating QC MatrixTable")
+
+    qc_mt = filter_low_conf_regions(mt, filter_exome_low_coverage_regions=filter_exome_low_coverage_regions, high_conf_regions=high_conf_regions)
+    if adj_only:
+        qc_mt = filter_to_adj(qc_mt) # TODO: Make sure that this works fine before call rate filtering
+    qc_mt = filter_rows_for_qc(qc_mt, min_af, min_callrate, inbreeding_coeff_threshold, apply_hard_filters)
+    if ld_r2 is not None:
+        pruned_ht = hl.ld_prune(qc_mt.GT, r2=ld_r2)
+        qc_mt = qc_mt.filter_rows(hl.is_defined(pruned_ht[qc_mt.row_key]))
+
+    qc_mt = qc_mt.annotate_globals(
+        qc_mt_params=hl.struct(
+            adj_only=adj_only,
+            min_af=min_af,
+            min_callrate=min_callrate,
+            inbreeding_coeff_threshold=inbreeding_coeff_threshold,
+            apply_hard_filters=apply_hard_filters,
+            ld_r2=ld_r2 if ld_r2 is not None else hl.null(hl.tfloat32),
+            filter_exome_low_coverage_regions=filter_exome_low_coverage_regions,
+            high_conf_regions=high_conf_regions if ld_r2 is not None else hl.null(hl.tarray(hl.tstr))
+        )
+    )
+    return qc_mt.annotate_cols(sample_callrate=hl.agg.fraction(hl.is_defined(qc_mt.GT)))
+
+
+def compute_callrate_mt(
+        mt: hl.MatrixTable,
+        intervals_ht: hl.Table = hl.import_locus_intervals(exome_calling_intervals_path),
+        bi_allelic_only: bool = True,
+        autosomes_only: bool = True
+) -> hl.MatrixTable:
+    """
+    Computes a sample/interval MT with each entry containing the call rate for that sample/interval.
+    This can be used as input for imputing exome sequencing platforms.
+
+    :param MatrixTable mt: Input MT
+    :param Table intervals_ht: Table containing the intervals. This table has to be keyed by locus. Default is the Broad exome calling intervals.
+    :param bool bi_allelic_only: If set, only bi-allelic sites are used for the computation
+    :param bool autosomes_only: If set, only autosomal intervals are used.
+    :return: Callrate MT
+    :rtype: MatrixTable
+    """
+    logger.info('Computing call rate MatrixTable')
+    if autosomes_only:
+        callrate_mt = filter_to_autosomes(mt)
+
+    if bi_allelic_only:
+        callrate_mt = callrate_mt.filter_rows(bi_allelic_expr(callrate_mt))
+
+    callrate_mt = callrate_mt.annotate_rows(interval=intervals_ht[callrate_mt.locus][intervals_ht.key])
```

Comment:
I wanted to annotate with whatever key the interval table has. Turns out this did not work, because when you index a table, you actually  don't get access to the table keys. So I had to write a bit of an odd workaround first annotating the intervals table with its own key in a different field and then using that. It's tested and works!